### PR TITLE
fix: ftl dev failing from go.mod files

### PR DESCRIPTION
**Pull request by wesbillman**

### CODE REVIEW

File: examples/go/echo/go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.23.0
 
 replace github.com/TBD54566975/ftl => ../../..
 
-require github.com/TBD54566975/ftl v0.248.0
+require github.com/TBD54566975/ftl v0.0.0-00010101000000-000000000000
```
Comment (wesbillman) [inline examples/go/echo/go.mod:7]: @alecthomas I wasn't 100% sure on this, but noticed that `echo` and `time` had different versions here so I made all of them the same. 😬

File: smoketest/relay/go.mod [hunk 2]
```diff
@@ -45,4 +46,4 @@ require (
 	google.golang.org/protobuf v1.34.2 // indirect
 )
 
-replace github.com/TBD54566975/ftl => /Users/safeer/dev/ftl
+replace github.com/TBD54566975/ftl => ../..
```
Comment (wesbillman) [inline smoketest/relay/go.mod:49]: This was the main change/fix for this PR

Comment (safeer) [reply smoketest/relay/go.mod:49]: 😬 thanks Wes

File: examples/go/echo/go.mod (hunk 1 above)
Comment (alecthomas) [reply examples/go/echo/go.mod:7]: If a correct `replace` directive exists, the version is irrelevant, so it doesn't hurt.

**Pull request merged**